phac-nml/irida
# fix-locked-samples-on-project-create

## Description of changes
What did you change in this pull request?  Provide a description of files changed, user interactions changed, etc.  Include how to test your changes.

Added functionality for a project owner to either leave samples unlocked or let them lock the samples from modification from other users (both managers and collaborators) that are members of the project. This is accomplished by adding a "Lock sample modification" checkbox to the Create New Project page

**To test locked samples:**
To test this functionality add samples to the cart and create a new project from these samples. On the create a project page select the "Lock sample modification" checkbox. Create the project. Add another user and add them to this project with a manager project role. Log in as this new user and go to the projects page and view the project that this user was added to. Click on any of the samples in the project which leads you to the sample details page. You should not have the ability to edit this sample.

**To test unlocked samples:**
To test this functionality add samples to the cart and create a new project from these samples. On the create a project page leave the "Lock sample modification" checkbox unchecked. Create the project. Add another user and add them to this project with a manager project role. Log in as this new user and go to the projects page and view the project that this user was added to. Click on any of the samples in the project which leads you to the sample details page. You should have the ability to edit this sample.

**Note:**
To test locked and unlocked samples for a collaborator follow the directions above but when adding the user to the project select **collaborator** for the project role. In either case a collaborator should **not** be allowed to edit any samples

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

Fixes #183

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:
